### ISSUE TITLE
MO2 fails to follow through the installation process

### ISSUE THREAD

**Issue by SamuelGogarty** [rockerbacon/modorganizer2-linux-installer#530]
I have Oblivion installed on my boot drive

I run ./install.sh

oblivion-mods/ > ./install.sh
INFO: all dependencies met
INFO: selected game 'oblivion'
INFO: found Steam in '/home/kaiser/.local/share/Steam'
INFO: game not found in '/home/kaiser/.local/share/Steam'
INFO: found Steam in '/home/kaiser/.var/app/com.valvesoftware.Steam/.local/share/Steam'
INFO: game not found in '/mnt/nvme1/SteamLibrary'
INFO: game not found in '/mnt/nvme2/SteamLibrary'
INFO: game not found in '/mnt/satassd/SteamLibrary'
INFO: game not found in '/mnt/HDD/SteamLibrary'
INFO: game found in '/home/kaiser/.var/app/com.valvesoftware.Steam/.local/share/Steam'
[kaiser@NeoTsar]
oblivion-mods/ >

as soon as I select oblivion and press next, the dialog closes and the script closes, no further dialogs.

**Comment by SamuelGogarty** [comment]
Also, the script seems to actually successfully continue to the next step if I have the game installed on a different drive. but that is not an option because MO2 fails to start when launching the game if the game is installed on a different drive.

**Comment by K1ngst0m** [comment]
same issue on steam deck:

```
(deck@steamdeck mo2installer-4.4.0)$ ./install.sh 
INFO: all dependencies met
INFO: selected game 'newvegas'
INFO: found Steam in '/home/deck/.local/share/Steam'
INFO: game found in '/home/deck/.local/share/Steam'
INFO: steam not found in '/home/deck/.var/app/com.valvesoftware.Steam/.local/share/Steam'
(1)(deck@steamdeck mo2installer-4.4.0)$ 
```

**Comment by PiersonCourtois** [comment]
Pop!_OS 22.04

I was able to fix this issue by installing the flatpak protontricks. I had originally downloaded it using  ```sudo apt install protontricks```, but when trying to open it it failed to launch. Uninstalling that and downloading the flatpak protontricks did the trick for me

**Comment by egrechko** [comment]
I have the same issue. I am running arch. Tried protontricks, protontricks-git, flatpak proton tricks. Installer will not get passed the detected game. 

### with protontricks-git

INFO: all dependencies met
INFO: selected game 'oblivion'
INFO: found Steam in '/home/e/.var/app/com.valvesoftware.Steam/.local/share/Steam'
INFO: game found in '/home/e/.var/app/com.valvesoftware.Steam/.local/share/Steam'
INFO: found Steam in '/home/e/.var/app/com.valvesoftware.Steam/.local/share/Steam'


### With flatpak

INFO: all dependencies met
INFO: selected game 'oblivion'
INFO: found Steam in '/home/e/.var/app/com.valvesoftware.Steam/.local/share/Steam'
INFO: game found in '/home/e/.var/app/com.valvesoftware.Steam/.local/share/Steam'
INFO: found Steam in '/home/e/.var/app/com.valvesoftware.Steam/.local/share/Steam'
protontricks (WARNING): Unknown Flatpak file system permission 'xdg-config/MangoHud:ro', ignoring.

**Comment by egrechko** [comment]
Well looks like I got it to install. 

Steps I took:
- Updated system (last update was a couple days ago)
- Installed steamtinkerlaunch
- After installing steamtinkerlaunch thought "aye why not give it another shot"
- Installer went through without issues.
- Uninstalled steamtinkerlaunch

**Comment by rockerbacon** [comment]
Just to give some update on the issue: this is currently non-actionable because the root cause is unknown and the error is not reproducible.
If someone affected by the issue could upload the terminal output for the command `set -x; ./install.sh; set +x`, we would be able to see the last thing the installer executes before it fails.

**Comment by FoxinTale** [comment]
> Just to give some update on the issue: this is currently non-actionable because the root cause is unknown and the error is not reproducible. If someone affected by the issue could upload the terminal output for the command `set -x; ./install.sh; set +x`, we would be able to see the last thing the installer executes before it fails.

As I've got the issue, here it goes. ```
(deck@steamdeck mo2installer-4.4.0)$ set -x; ./install.sh; set+x;
+ set -x
+ ./install.sh
INFO: all dependencies met
INFO: selected game 'skyrimspecialedition'
INFO: found Steam in '/home/deck/.local/share/Steam'
INFO: game found in '/home/deck/.local/share/Steam'
INFO: steam not found in '/home/deck/.var/app/com.valvesoftware.Steam/.local/share/Steam'
+ set+x
bash: set+x: command not found
++ __steamos_prompt_command '(\[\033[1;32m\]\u@\h\[\033[1;34m\] \W\[\033[0m\])\$ '
++ local rc=127
++ local 'ps1=(\[\033[1;32m\]\u@\h\[\033[1;34m\] \W\[\033[0m\])\$ '
++ local partset
++ ischroot
+++ diff -q /proc/1/mountinfo /proc/self/mountinfo
++ test -n ''
++ [[ 127 -ne 0 ]]
++ ps1='(\[\033[31;1m\]127\[\033[0m\])(\[\033[1;32m\]\u@\h\[\033[1;34m\] \W\[\033[0m\])\$ '
++ PS1='(\[\033[31;1m\]127\[\033[0m\])(\[\033[1;32m\]\u@\h\[\033[1;34m\] \W\[\033[0m\])\$ '
++ return 127
(127)(deck@steamdeck mo2installer-4.4.0)$
```

I have some Linux experience but I have uh...no clue what this means, or if it's even useful to you. But it is the result of that output.

**Comment by rockerbacon** [comment]
My bad, the correct command is `set -x; source ./install.sh; set +x`

**Comment by FoxinTale** [comment]
> My bad, the correct command is `set -x; source ./install.sh; set +x`

It closes the terminal shortly after selecting the option, so I am unable to capture and copy/paste the entire output but I was able to get a screenshot of the bottom half of the output. I'm unsure how useful what I was able to get is, but I'll update the comment if I'm able to snag the full output later. 

![Screenshot_20240124_230448](https://github.com/rockerbacon/modorganizer2-linux-installer/assets/51178662/393c401a-2ad7-4a5f-b294-5be213cce985)

**Comment by rockerbacon** [comment]
Ok, we have a sign of where the script is failing now. Could you give me the output for the command:
```
flatpak run 'com.github.Matoking.protontricks' -c 'echo $WINEPREFIX' 489830 && echo Success || echo Failure
```

**Comment by FoxinTale** [comment]
> Ok, we have a sign of where the script is failing now. Could you give me the output for the command:
> 
> ```
> flatpak run 'com.github.Matoking.protontricks' -c 'echo $WINEPREFIX' 489830 && echo Success || echo Failure
> ```

This returns "success" However, I followed a hunch I described in my previous comment and ran the game once, getting to the main menu. And then ran the install script. After this, all is now good and it works. 

I was actually able to replicate this issue with success in testing.  While I am not sure if  this is true for others who may be having the issue, it seems that if the `steamapps/compatdata/489830` (or the related appID) folder exists, but is totally empty the installer is not sure how to handle it. 

The given command to try, when I had an empty prefix folder returned "failure". 

The fix for my issue is to run the game, from my testing. As I stated in the last comment, I uninstalled the game from my microSD (after three failed move attempts) and re-installed it on the internal drive, and for some reason assumed the prefix would carry over. That turned out to be a silly (and incorrect) assumption.
